### DISCUSSION TITLE
c4fm

### DISCUSSION THREAD

**Discussion by misahoskova** [egzumer/uv-k5-firmware-custom#357] [Q&A] [answered]
Hello, 
I have a one question about digital modulations. Would it be possible on this radio implemented C4FM?
Thank you for your answer.

**Comment by OE1MWW** [comment]
no way, memory is full, don't forget, this is a 20$ device

**Reply by misahoskova** [reply]
But the problem is only memory? What if there will be some feature removed, for example spectrum scope, will there be some space for another modulation?

**Comment by egzumer** [comment]
This is an analog FM transceiver. Don't expect miracles.

**Comment by misahoskova** [comment]
Of course, it is cheap hw but is it worth trying?

**Reply by egzumer** [reply, marked as answer]
Feel free to investigate:
[bk4819.zip](https://github.com/egzumer/uv-k5-firmware-custom/files/13799119/bk4819.zip)

**Reply by misahoskova** [reply]
Thanks ♥️

**Reply by misahoskova** [reply]
How is achieved that the signal processor BK4819 can receive AM? Because in all data sheets is only mentioned FM.

**Reply by egzumer** [reply]
AM uses one of the undocumented settings in AF mode register.

**Reply by misahoskova** [reply]
Ok, same for usb? But there is no programable signal processor? Problem is we are stuck on hardware features of bk4819. So if I would not be able to find magic hidden feature for frequency keying and decoding, there is no way to implement C4FM?

**Reply by egzumer** [reply]
I already told you all I know. Simple answer is it is impossible. Can you find some way around? I don't know.

**Reply by misahoskova** [reply]
Maybe some of this mode can be another modulation. do you know who tested?

driver/bk4819.h
```
enum BK4819_AF_Type_t
{
	BK4819_AF_MUTE      =  0u,  //
	BK4819_AF_FM        =  1u,  // FM
	BK4819_AF_ALAM      =  2u,  //
	BK4819_AF_BEEP      =  3u,  //
	BK4819_AF_BASEBAND1 =  4u,  // RAW
	BK4819_AF_BASEBAND2 =  5u,  // USB
	BK4819_AF_CTCO      =  6u,  // strange LF audio .. maybe the CTCSS LF line ?
	BK4819_AF_AM        =  7u,  // AM
	BK4819_AF_FSKO      =  8u,  // nothing
	BK4819_AF_UNKNOWN3  =  9u,  // BYP
	BK4819_AF_UNKNOWN4  = 10u,  // nothing at all
	BK4819_AF_UNKNOWN5  = 11u,  // distorted
	BK4819_AF_UNKNOWN6  = 12u,  // distorted
	BK4819_AF_UNKNOWN7  = 13u,  // interesting
	BK4819_AF_UNKNOWN8  = 14u,  // interesting
	BK4819_AF_UNKNOWN9  = 15u   // not a lot
};
```

8u should be FSK. same as virgin also in application note.

**Reply by egzumer** [reply]
No idea
